engine/buildkit/telemetry.go

```diff
@@ -41,19 +40,23 @@ type buildkitTracer struct {
 }
 
 func (t *buildkitTracer) Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
```

Comment:
I wonder if there's a clean way to do this as part of our SpanProcessor pipeline instead - it might be more robust than relying on Buildkit using `ctx`. Not a blocker, just thinking out loud. Vaguely thinking something like identifying a span's instrumentation library ("this came from buildkit") and then directly modifying it in `OnStart` where we have write access. But I'm not sure if there's enough identifying information.

Reply:
> "this came from buildkit"

I think this is difficult right now:

https://github.com/moby/buildkit/blob/75bbf7b2071ef8b1d8dff70d4d3d878a53c6d991/util/tracing/tracing.go#L25-L27

This is where you're supposed to put a nice little uri right? We put `dagger.io/...` here, maybe some more upstream PRs are called for :cry:

Reply:
Haha yep, just ran into that myself. I think it'd be worth it. Otherwise it just defaults to the Go Otel SDK instrumentation library value I think.